python/lsst/pipe/tasks/calibrate.py

```diff
@@ -255,17 +281,25 @@ def __init__(self, icSourceSchema=None, **kwargs):
                     self.schemaMapper.addMapping(schemaItem.getKey())
 
             if missingFieldNames:
-                raise RuntimeError(
-                    "isSourceCat is missing fields %s specified in icSourceFieldsToCopy" %
-                     (missingFieldNames,))
+                raise RuntimeError("isSourceCat is missing fields {} specified in icSourceFieldsToCopy"
+                                   .format(missingFieldNames))
 
             # produce a temporary schema to pass to the subtasks; finalize it later
             self.schema = self.schemaMapper.editOutputSchema()
         else:
             self.schemaMapper = None
             self.schema = afwTable.SourceTable.makeMinimalSchema()
+        self.makeSubtask('detection', schema=self.schema)
+
+        self.algMetadata = dafBase.PropertyList()
+
+        if self.config.doDeblend:
+            self.makeSubtask("deblend", schema=self.schema)
+        self.makeSubtask('measurement', schema=self.schema, algMetadata=self.algMetadata)
+        if self.config.doApCorr:
+            self.makeSubtask('applyApCorr', schema=self.schema)
+        self.makeSubtask('afterburners', schema=self.schema)
```

Comment:
I meant to talk to you about this while you were here. Jim and I discussed it at length with Russel. The problem is that there are many cases where you do very similar but not the same operations with these tasks. Having them all together then necessitated having a complex series of config options and function parameters to control the execution flow. We decided a bit of duplication was acceptable to simplify the execution logic. I have a proposal for a type of mixin that would reduce the boiler plate for much of this at some point in the future. It might be worth talking about at the Aug meeting. If it is ok with you, the verbosity will remain for this ticket, and will hopefully be addressed in the furture.